include/crocoddyl/multibody/costs/control-gravity-contact.hxx

```diff
@@ -89,14 +89,21 @@ void CostModelControlGravContactTpl<Scalar>::calcDiff(
 
   activation_->calcDiff(data->activation, data->r);
   
-  data->Lu.noalias() =  data->activation->Ar; 
-  data->Lx.noalias() =  -d->dg_dx.rightCols(nu_) * data->activation->Ar; 
+  for (typename ContactModelMultiple::ContactDataContainer::iterator it = d->contacts.begin();
+         it != d->contacts.end(); ++it) {
+			 d->df_du_Jc.noalias() += (it->second->df_du.transpose() * it->second->Jc).rightCols(nu_);
```

Comment:
Retrieving the force derivatives like this gives a df_du and df_dx = 0, and I don't know why. How can I get the current df_dx and df_du?

Reply:
Are you enabling the force in the differential action model?
By default, the derivatives of the contact forces are not computed.

Reply:
I have set enable_force = true during the creation of the DifferentialActionModelContactFwdDynamics but when I print the df_dx and df_du it is again zero

Reply:
You have an issue with the typedef (see above). I wonder how this code compiles!
Fix it and let me know

Reply:
I fixed the typedef and it compiles, only thing is the check against numerical differentiation is failing. I don't see how my computation of the derivatives is wrong, so I think Rohan's help would be greatly appreciated here. I'm suspecting that the derivatives of the forces are not needed, since when I remove them the numerical diff tests are passing; however, theoretically speaking, f should depend on x and u and therefore its derivatives should play a role.